### ISSUE TITLE
Keystore decryption results are strange

### ISSUE THREAD

**Issue by yulidai** [prysmaticlabs/prysm#6869]
# 💎 Keystore decryption results are strange

### Background

Version: v1.0.0-alpha.18

### Description
The Keystore I did not generate through the official tool is as follows:
```
{"name": "keystore", "uuid": "ed98ea7f-e7ef-4326-a4ca-54d006bc747b", "crypto": {"kdf": {"params": {"n": 32768, "p": 1, "r": 8, "salt": "1fa4ea621659b11f5aca0966ed3273a759d754b5f4b529300356ad04067c012b", "dklen": 32}, "message": "", "function": "scrypt"}, "cipher": {"params": {"iv": "fdd8e56744b4883a5d260446eeddf68c"}, "message": "a8969e1b0e60dc0d45a0abdb1f3c9fe97834273f67510f0a33dc0c7dfde9870c", "function": "aes-128-ctr"}, "checksum": {"params": {}, "message": "525741fb9c756c7847a4beb1c7d7c9a8770edb177042a9b9d705370db6b32237", "function": "sha256"}}, "version": 4, "next_account": 1}
```
The public key obtained after unlocking with password `$qA0bZkCLxyWVeb8Q2^S~Crjs4^!eV1B` is `9498d58444dd7570506d41455129ca4e842625ad8e80f8c6eddb935d6bcb2ca7e16146b085a220af782b34d6c6663ec9`, but after the Prysm client is started, the log shows:
```
pubKey=0x87db1ad35ed1 status="UNKNOWN_STATUS"
```
I don't understand why? Has the client generated a new `Keypair` based on my `Keystore`?

The `keystore` is generated according to the crypto format by myself, and the generated result can be correctly parsed by Lighthouse. It can also be decrypted correctly in Prysm but a different public key is parsed.

 I doubt that prysm has fully tested `pdkdf2` but did not pay much attention to `scrypt`, although it supports

**Comment by yulidai** [comment]
@terencechain  Currently I encrypt the 32-byte private key directly. How should I convert these 32 bytes before encryption to be correctly recognized by prysm? Any help would be greatly appreciated. :P

**Comment by rauljordan** [comment]
Hi @yulidai we identified the issue. It will be resolved with #6914
